pkg/controller/awsfederatedaccountaccess/awsfederatedaccountaccess_controller.go

```diff
@@ -0,0 +1,368 @@
+
+var log = logf.Log.WithName("controller_awsfederatedaccountaccess")
+
+// Add creates a new AWSFederatedAccountAccess Controller and adds it to the Manager. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager) error {
+	return add(mgr, newReconciler(mgr))
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+	return &ReconcileAWSFederatedAccountAccess{
+		client: mgr.GetClient(),
+		scheme: mgr.GetScheme(),
+	}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New("awsfederatedaccountaccess-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to primary resource AWSFederatedAccountAccess
+	err = c.Watch(&source.Kind{Type: &awsv1alpha1.AWSFederatedAccountAccess{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return err
+	}
+
+	// watch for changes to secondary resource AWSFederatedRole
+	err = c.Watch(&source.Kind{Type: &awsv1alpha1.AWSFederatedRole{}}, &handler.EnqueueRequestForOwner{
+		IsController: true,
+		OwnerType:    &awsv1alpha1.AccountPool{},
+	})
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// blank assignment to verify that ReconcileAWSFederatedAccountAccess implements reconcile.Reconciler
+var _ reconcile.Reconciler = &ReconcileAWSFederatedAccountAccess{}
+
+// ReconcileAWSFederatedAccountAccess reconciles a AWSFederatedAccountAccess object
+type ReconcileAWSFederatedAccountAccess struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client client.Client
+	scheme *runtime.Scheme
+}
+
+// Reconcile reads that state of the cluster for a AWSFederatedAccountAccess object and makes changes based on the state read
+// and what is in the AWSFederatedAccountAccess.Spec
+func (r *ReconcileAWSFederatedAccountAccess) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling AWSFederatedAccountAccess")
+
+	// Fetch the AWSFederatedAccountAccess instance
+	currentFAA := &awsv1alpha1.AWSFederatedAccountAccess{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, currentFAA)
+	if err != nil {
+		if k8serr.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	// If the state is applied don't do anything
```

Comment:
What is 'applied'? This comment doesn't seem to match what you're checking, which is "Ready" or "Failed"